br/pkg/lightning/duplicate/detector.go

```diff
@@ -100,11 +100,17 @@ func (d *Detector) Detect(ctx context.Context, opts *DetectOptions) (numDups int
 		})
 	}
 
+	errCh := make(chan error, 1)
+	go func() {
+		errCh <- g.Wait()
+		for range taskCh {
```

Comment:
if `taskCh` is unbuffered channel, we don't have to drain the tasks 🤔

Reply:
Buffered `taskCh` is also used by one worker to check whether need to split and prepare tasks for other soon-to-be-idle workers.